Bio/SeqIO/SnapGeneIO.py

```diff
@@ -170,6 +167,16 @@ def _parse_features_packet(length, data, record):
                     qvalues.append(int(value.attributes["int"].value))
             quals[qname] = qvalues
 
+        name = _get_attribute_value(feature, "name")
+        if name:
+            if "label" not in quals:
+                # No explicit label attribute, use the SnapGene name
+                quals["label"] = [name]
+            elif name not in quals["label"]:
+                # The SnapGene name is different from the label,
```

Comment:
This works in the example case (where the name is longer than the label) but the other way round (e.g. name = "laqI", label = "lacI repressor") it would discard the name and only keeps the label. Am I right?

Reply:
Thus, we want to check for equality?

Reply:
We do want to check for equality, and that is what we are doing here. The ``not in`` test is not intended to search for a substring within the qualifier, but to search for a value within the list of values associated with the qualifier (remember that each qualifier contains not a single value but an *array* of values, even if most cases the array contains a single element).

Reply:
My bad, each qualifier is a _list_...

Reply:
Yes, this often catches people out, but it models the worse case of duplicated qualifier keys in GenBank/EMBL files.